### ISSUE TITLE
Nav dropdowns don't work on archive pages

### ISSUE THREAD

**Issue by diegovogel** [pojome/elementor#3978]
I'm using Elementor to build my site header which includes a nav menu with some dropdowns for submenus. The dropdowns work fine on all static and single-post pages (including blog posts and woocommerce products), but they don't work at all on archive pages such as the blog list or any woocommerce category pages. 

This error appears in the Dev Tools console on the pages where the dropdowns don't work:
```
Uncaught TypeError: Cannot read property 'elementor_stretched_section_container' of undefined
    at n (frontend.min.js?ver=2.0.3:2)
    at Object.getGeneralSettings (frontend.min.js?ver=2.0.3:2)
    at i.stretchSection (frontend.min.js?ver=2.0.3:2)
    at i.function.a.(/racingandriding/blog/anonymous function) [as stretchSection] (http://sites.btwebgroup.com/racingandriding/wp-content/plugins/elementor/assets/js/frontend.min.js?ver=2.0.3:2:34105)
    at i.onInit (frontend.min.js?ver=2.0.3:2)
    at i.function.a.(/racingandriding/blog/anonymous function) [as onInit] (http://sites.btwebgroup.com/racingandriding/wp-content/plugins/elementor/assets/js/frontend.min.js?ver=2.0.3:2:34105)
    at i.trigger (frontend.min.js?ver=2.0.3:2)
    at i.function.a.(/racingandriding/blog/anonymous function) [as trigger] (http://sites.btwebgroup.com/racingandriding/wp-content/plugins/elementor/assets/js/frontend.min.js?ver=2.0.3:2:34105)
    at i.r (frontend.min.js?ver=2.0.3:2)
    at i (frontend.min.js?ver=2.0.3:2)
```
The dropdowns are still present in the page source code, but they don't activate when I click on the appropriate menu items and the chevrons next to those items are also missing.

I'm using Elementor Pro 1.15.6 with the Astra theme, and I used the "Header Footer Elementor" plugin to build a site-wide header with Elementor. 

I don't know enough JS to decode that error, so any help is much appreciated. If this isn't the right place to post this bug I'll gladly post it elsewhere. Thanks!

**Comment by diegovogel** [comment]
Here's my system info, from Elementor:
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: 5.6.38
	PHP Version: 5.6.34
	PHP Max Input Vars: 1000
	PHP Max Post Size: 8M
	GD Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 4.9.4
	Site URL: http://sites.btwebgroup.com/racingandriding
	Home URL: http://sites.btwebgroup.com/racingandriding
	WP Multisite: No
	Max Upload Size: 8 MB
	Memory limit: 40M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Astra
	Version: 1.1.6
	Author: Brainstorm Force
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 2.0.3
		Author: Elementor.com

	Elementor Pro
		Version: 1.15.6
		Author: Elementor.com

	Header Footer Elementor
		Version: 1.0.12
		Author: Brainstorm Force, Nikhil Chavan

	WooCommerce
		Version: 3.3.4
		Author: Automattic


== Debug ==
	Errors: 
		ViewDestroyedError: View (cid: "view4675") has already been destroyed and cannot be used.
			at http://sites.btwebgroup.com/racingandriding/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5 - 22:19954
			2018-03-31 19:48 +00:00

		TypeError: Cannot read property 'errors' of undefined
			at http://sites.btwebgroup.com/racingandriding/wp-content/plugins/elementor/assets/js/editor.min.js?ver=1.9.3 - 2:113818
			2018-01-30 21:23 +00:00
			x 13 times

		SyntaxError: Unexpected token ;
			at http://sites.btwebgroup.com/racingandriding/wp-admin/post.php?post=79&action=elementor - 2:91
			2018-01-30 21:10 +00:00

		SyntaxError: Unexpected token ;
			at http://sites.btwebgroup.com/racingandriding/wp-admin/post.php?post=79&action=elementor - 2:90
			2018-01-30 21:10 +00:00

		SyntaxError: Unexpected token new
			at http://sites.btwebgroup.com/racingandriding/wp-admin/post.php?post=79&action=elementor - 2:61
			2018-01-30 21:10 +00:00

		SyntaxError: Unexpected token new
			at http://sites.btwebgroup.com/racingandriding/wp-admin/post.php?post=79&action=elementor - 2:60
			2018-01-30 21:10 +00:00

		SyntaxError: Invalid or unexpected token
			at http://sites.btwebgroup.com/racingandriding/wp-admin/post.php?post=79&action=elementor - 2:52
			2018-01-30 21:10 +00:00
			x 6 times

		SyntaxError: Unexpected token ;
			at http://sites.btwebgroup.com/racingandriding/wp-admin/post.php?post=79&action=elementor - 2:52
			2018-01-30 21:09 +00:00

		TypeError: Cannot set property 'innerHTML' of null
			at http://sites.btwebgroup.com/racingandriding/wp-admin/post.php?post=79&action=elementor - 2:55
			2018-01-30 21:09 +00:00
			x 4 times

		ReferenceError: copyright is not defined
			at http://sites.btwebgroup.com/racingandriding/wp-admin/post.php?post=79&action=elementor - 2:29
			2018-01-30 20:56 +00:00
			x 4 times

**Comment by bainternet** [comment]
@diegovogel 

If it all works fine on "static" (regular) pages then this seems to be related to "Header Footer Elementor", please contact the plugin author.

thanks.

**Comment by diegovogel** [comment]
Ok will do, thanks.

**Comment by diegovogel** [comment]
@bainternet I [posted this issue](https://github.com/Nikschavan/header-footer-elementor/issues/99) to the Header Footer Elementor github page and they said it's unlikely the plugin is at fault because they don't add any JS to the frontend. Any thoughts?

**Comment by robertmorel** [comment]
Agreed. I Just received a ticket for my company website saying the menu on the archive pages is broken

**Comment by robertmorel** [comment]
Same error message

**Comment by twotweak** [comment]
I have the same problem, using Elementor (building the site header nav menu) + Astra Pro (for custom layouts for header and footer). I find out that, when I deactivate the "Astra Pro" plugin, the dropdown/submenu shows and works on the (woocommerce product/category) archive pages.

I have submitted a support ticket to Astra Support (waiting for reply) ...

**Comment by diegovogel** [comment]
@twotweak please let us know if you hear anything back from Astra Support.
I just tried the OceanWP theme on my site and had the same exact issue. I also tried the default Wordpress Twenty Seventeen theme and the dropdowns worked on all pages but this theme isn't supported by Header Footer Elementor so other parts of the header/footer were broken.

**Comment by diegovogel** [comment]
@robertmorel @twotweak -- if you turn off "Stretch Section" for all the sections in the header and footer templates this error will go away.

![image](https://user-images.githubusercontent.com/23545422/38307459-d5c5e97c-37e1-11e8-99d2-73cbb07e881c.png)

This was not an issue before, so I don't know what changed... but turning off "Stretch Section" doesn't seem to have any visual affect on the header and footer so this should work.

**Comment by twotweak** [comment]
@diegovogel -- thanks for finding the solution. I have just turned off the "stretch section" and it worked, nice!

**Comment by robertmorel** [comment]
Thanks @diegovogel  this worked for me. 

For anyone interested a css hack for elementor stretched row is to give the row an Id and then give the row css  min-width:100% !important;

**Comment by gabetu** [comment]
Same problem with OceanWP theme, Nicolas saying that it's an Elementor issue (https://oceanwp.org/whats-new-oceanwp-mobile-header-opengraph/#comment-24576).

``== Debug ==
	Errors: 
		TypeError: An invalid argument supplied as multiple control value
			at https://www.notarpiatraneamt.ro/wp-content/plugins/elementor/assets/js/editor.min.js?ver=1.9.7 - 2:86363
			2018-03-09 15:53 +00:00
			category: InvalidElementData
			tag: Element `undefined` got <> as `pin_image` value. Expected array or object.

		TypeError: j is undefined
			at https://www.notarpiatraneamt.ro/wp-admin/js/editor.min.js?ver=f82a06f54f4fce934c2c2bd5d1946f4e - 1:681
			2018-03-04 14:41 +00:00
			x 9 times

**Comment by redpik** [comment]
Same bug for me.

`Uncaught TypeError: Cannot read property 'elementor_stretched_section_container' of undefined`

Elementor 2.0.7
Elementor Pro 1.9.3

Elementor Pro updated : 2.0.2 version, but same bug.

**Comment by gabetu** [comment]
Deactivating, deleting and reinstalling latest version of Elementor made the problem go away for me.

**Comment by robertmorel** [comment]
CSS hacks - 

Option 1 : Turn off stretched section
Option 2 : Give the row a class and set the row width to 100%
Option 3: Give the row a class and on screen width less than 800px set the width of the row to 104.5% with margin-left: -3%

**Comment by Seloha** [comment]
Still a problem.

**Comment by diegovogel** [comment]
@Seloha  did you try turning off Stretch Section for all header sections?
https://github.com/elementor/elementor/issues/3978#issuecomment-378582347

**Comment by Seloha** [comment]
@diegovogel Yeah I checked that. None of the sections were stretched. The problem completely disappeared once I changed the theme to Phlox.

**Comment by diegovogel** [comment]
Gotcha, glad you got it fixed. Out of curiosity, are you using the Header Footer Elementor plugin or the Elementor Theme Builder to create the header?

**Comment by Seloha** [comment]
I am using Elementor Pro to build it. Haven't tried that plugin.

**Comment by diegovogel** [comment]
I see. This issue was originally posted before Elementor Pro included the Theme Builder, so the Header Footer plugin was necessary. It's no longer needed now that Elementor Pro has the Theme Builder.

**Comment by coryhubbell** [comment]
I am still suffering with drop downs in woocommerce

**Comment by jackithomas** [comment]
Same here. No drop downs on Woocommerce pages.

**Comment by cryptonyx2** [comment]
same here, and now it is almost march 2020. no dropdowns on my nav menu on woocommerce product pages. Both desktop classic and mobile hamburger menu. header builth with elementor theme builder

**Comment by mopquill** [comment]
Still an issue on Elementor 2.9.8 and Elementor Pro 2.9.3. Menus with dropdowns appear to lose them on WooCommerce pages. We're not using that Header Footer plugin mentioned above; it's disappointing to see this issue being ignored going into May 2020. It's disappointing to see that this issue has been ignored.

@bainternet Is it possible to reopen this issue, seeing it's ongoing? I've double-checked that [no other open bugs](https://github.com/elementor/elementor/issues?q=is%3Aissue+is%3Aopen+dropdown) (from reading titles) seem to match, and I found this issue from Google while searching it.

We're not using plugins mentioned in some of those other issues like Autoptimize, and we're on the Massive theme.

**Comment by avihaip** [comment]
OK. had the same problem.
i think the problem comes from the intefration between theme and elementor pro.
changed theme to HELLO ELEMENTOR and the problem solved.
i have dropdown menus icon on archive and product pages.

**Comment by mopquill** [comment]
Can confirm. The issue is really that the dropdown menus are added in via JavaScript. In my case, there was an outdated bootstrap.min.js (3.x, and not even the latest version) in the Massive theme (I have since notified them of this breaking Elementor dropdowns) causing a JavaScript error. So the function was crashing out, and no menus being generated.

I'm not wild about the menus depending on JavaScript, but writing a small function to dequeue/deregister the bootstrap.min.js from the massive child theme's functions.php, and enqueuing an up-to-date bootstrap.min.js worked fine. If anyone is using Massive and needs me to, I can provide the code.

**Comment by cstrouse** [comment]
This is broken using Elementor Pro and their own Hello theme, too.

**Comment by mopquill** [comment]
Check for outdated bootstrap in the hello theme and throw in a function to replace it an up-to-date one. I'd attach one here, but it varies based on the theme. You just want to dequeue/register the .js and enqueue your own with the same requirements to roughly keep the stack order. You can find the requirements by looking through their source to find how they enqueue it.
